### PR TITLE
csvtk: init at 0.30.0

### LINKED ISSUE #328735: Package request: csvtk (Go)
**Project description**

> A cross-platform, efficient and practical CSV/TSV toolkit in Golang

> People usually use spreadsheet software like MS Excel to process
> table data. However this is all by clicking and typing, which is not
> automated and is time-consuming to repeat, especially when you want
> to apply similar operations with different datasets or purposes.
>
> You can also accomplish some CSV/TSV manipulations using shell
> commands, but more code is needed to handle the header line. Shell
> commands do not support selecting columns with column names either.
>
> csvtk is convenient for rapid data investigation and also easy to
> integrate into analysis pipelines. It could save you lots of time in
> (not) writing Python/R scripts.

**Metadata**

* homepage URL: https://bioinf.shenwei.me/csvtk/
* source URL: https://github.com/shenwei356/csvtk
* license: mit
* platforms: unix

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by luftmensch-luftmensch**
## Description of changes
Closes #328735

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by pacien — APPROVED** [review]
Thanks for packaging this!

Diff looks good to me;
Quickly tested successfully with:

```fish
curl https://jsvine.github.io/intro-to-visidata/_downloads/a61d9b28e9a942e1254bffeb8289a447/faa-wildlife-strikes.csv |
  nix run \
    "github:NixOS/nixpkgs?rev=6f4c5400914bb10ae8c1f971873449a541a9e516#csvtk" \
	-- summary -f ATYPE:countuniq
```

**Comment by smancill** [comment]
Result of `nixpkgs-review pr 328866` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>csvtk</li>
  </ul>
</details>

File: pkgs/by-name/cs/csvtk/package.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+  ...
+}:
```
Comment (smancill) [inline pkgs/by-name/cs/csvtk/package.nix:6]:
```suggestion
}:
```

File: pkgs/by-name/cs/csvtk/package.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+let
+  version = "0.30.0";
+in
+buildGoModule {
+  pname = "csvtk";
+  inherit version;
```
Comment (smancill) [inline pkgs/by-name/cs/csvtk/package.nix:12]:
```suggestion
buildGoModule rec {
  pname = "csvtk";
  version = "0.30.0";
```

Comment (smancill) [reply pkgs/by-name/cs/csvtk/package.nix:12]: Not sure about this one, though, but it seems to me the version using `rec` is more common.

Comment (luftmensch-luftmensch) [reply pkgs/by-name/cs/csvtk/package.nix:12]: I tought that we are ditching the use of `rec` in favour of the `let in` pattern

Comment (smancill) [reply pkgs/by-name/cs/csvtk/package.nix:12]: Oh, OK, I wasn't aware. So many new rules.

File: pkgs/by-name/cs/csvtk/package.nix (hunk 1 above)
Comment (smancill) [reply pkgs/by-name/cs/csvtk/package.nix:6]: Could you update this so I can approve? I've seen other PRs where it is also asked, and just a handful of packages has it like this.

Comment (luftmensch-luftmensch) [reply pkgs/by-name/cs/csvtk/package.nix]: @smancill  Done!

**Comment by smancill** [comment]
Result of `nixpkgs-review pr 328866` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>csvtk</li>
  </ul>
</details>

**Review by smancill — APPROVED** [review]

**Pull request merged**